console-subscriber/src/callsites.rs

```diff
@@ -0,0 +1,53 @@
+use std::{
+    ptr,
+    sync::atomic::{AtomicPtr, AtomicUsize, Ordering},
+};
+use tracing_core::Metadata;
+
+#[derive(Debug, Default)]
+pub(crate) struct Callsites {
+    ptrs: [AtomicPtr<Metadata<'static>>; MAX_CALLSITES],
```

Comment:
I wonder, is the plan to use this structure for tracking other more granular callsites such as resource, resource ops? If that is the case, the number can grow to a value larger than 1 - 5. Is the plan to switch to a map based approach then or would a linear scan still be _fast enough_

Reply:
I was thinking we'd solve that problem when it becomes necessary :) 

I think we may just want to use a hashset (or something) for callsites we expect to see a lot of, and arrays for callsites we expect there to be fewer of --- we *could* come up with some kind of adaptive structure that switches dynamically once it reaches a certain length, but in this case, I think we can make pretty good guesses about which callsites we expect to see a lot of, and which we expect to see fewer of.

Reply:
Also, as we start caring about more types of callsites, I think we may want to switch to a model where we store the callsite alongside some data about which type of callsite it is, in one array, instead of storing each type of callsites in separate arrays. That way, we only do _one_ linear scan, and then know everything we need to about that callsite, instead of having to look in multiple arrays.